src/jit/codegenxarch.cpp

```diff
@@ -6184,6 +6201,14 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
 #endif // defined(_TARGET_X86_)
 
+    if (call->IsTailCallViaHelper())
+    {
+        if (compiler->getNeedsGSSecurityCookie())
+        {
+            genEmitGSCookieCheck(true);
+        }
+    }
```

Comment:
I put it under `#ifdef _TARGET_X86_`. Turns out I think it doesn't matter because AMD64 would never set to do GS cookie and tailcall.